src/cephadm/node-proxy/reporter.py

```diff
@@ -23,6 +26,19 @@ def loop(self):
             # scenario probably we should just send the sub-parts
             # that have changed to minimize the traffic in
             # dense clusters
-            d = self.system.get_system()
-            requests.post(self.observer_url, json=d)
-            time.sleep(10)
+            if self.system.data_ready:
+                log.debug("waiting for a lock.")
+                try:
+                    self.system.lock.acquire()
+                    log.debug("lock acquired.")
+                    if not self.system.get_system() == self.system.previous_data:
```

Comment:
would this compare the whole dictionary? ... I think for performance we have to think about calculating some hash and use it for the check to avoid comparing all the data which could take some time.

Reply:
I think it compares the whole dict (not 100% sure though). But yeh, this is a discussion I've had with @adk3798.
In any case, ideally it would be nice to get the difference and only send that to the endpoint.